Source/WebGPU/WebGPU/BindGroup.mm

```diff
@@ -50,127 +50,27 @@ static bool textureViewIsPresent(const WGPUBindGroupEntry& entry)
     return entry.textureView;
 }
 
-#if HAVE(TIER2_ARGUMENT_BUFFERS)
-static auto sizeOfEntries(const WGPUBindGroupDescriptor& descriptor, BindGroupLayout* bindGroupLayout)
-{
-    uint32_t sizes[] = { 0, 0, 0 };
-    for (uint32_t entryIndex = 0; entryIndex < descriptor.entryCount; ++entryIndex) {
-        const WGPUBindGroupEntry& entry = descriptor.entries[entryIndex];
-
-        auto stages = bindGroupLayout ? bindGroupLayout->stagesForBinding(entry.binding) : (WGPUShaderStage_Vertex | WGPUShaderStage_Fragment | WGPUShaderStage_Compute);
-        constexpr WGPUShaderStage shaderStage[] = { WGPUShaderStage_Vertex, WGPUShaderStage_Fragment, WGPUShaderStage_Compute };
-        bool bufferIsPresent = WebGPU::bufferIsPresent(entry);
-        bool samplerIsPresent = WebGPU::samplerIsPresent(entry);
-        bool textureViewIsPresent = WebGPU::textureViewIsPresent(entry);
-
-        for (size_t currentStage = 0; currentStage < WTF_ARRAY_LENGTH(shaderStage); ++currentStage) {
-            WGPUShaderStage renderStage = shaderStage[currentStage];
-            if (!(stages & renderStage))
-                continue;
-
-            if (bufferIsPresent)
-                sizes[currentStage] += sizeof(float*);
-            else if (samplerIsPresent || textureViewIsPresent)
-                sizes[currentStage] += sizeof(MTLResourceID);
-        }
-    }
-
-    return std::array<uint32_t, 3>( { sizes[0], sizes[1], sizes[2] } );
-}
-#endif
-
 Ref<BindGroup> Device::createBindGroup(const WGPUBindGroupDescriptor& descriptor)
 {
     if (descriptor.nextInChain)
         return BindGroup::createInvalid(*this);
 
+    constexpr WGPUShaderStage shaderStage[] = { WGPUShaderStage_Vertex, WGPUShaderStage_Fragment, WGPUShaderStage_Compute };
```

Comment:
I'd like your opinion about something. On a previous review, Kimmo suggested considering the `WGPU` types as external types, only to be used at the API boundary of the framework. Just like how we have `WGPUBuffer` as an API type, but internally we have `WebGPU::Buffer`, we could extend that relationship to _all_ `WGPU` types. The "binding" code to turn all `WGPU` types into their corresponding `WebGPU::` types would be generated code, driven by IDL as a custom build step.

What do you think about the tradeoff of having cleaner internal/external boundaries, vs the added infrastructure (and potential runtime cost?) of transforming a bunch of types into different types that have the same expressivity?

Reply:
I think this makes a lot of sense since if the external (WGPU) types change in any form (renames or values change), we do not want to have a lot of compilation or runtime breakage throughout our code.

Reply:
I'll make this change prior to merging